### PR TITLE
fix: failed to start in k3s cluster

### PR DESCRIPTION
**Pull request by thxCode**
support Russian Dolls

**problem**

1. the embedded k3s IP conflicts with the on-host cluster
2. the nesting embedded k3s keeps using root cgroup, which causes launching to fail. 

**solution**

1. when it starts k3s, walrus tries to pick a proper subnet according to the host's subnet.
2. when it starts k3, walrus picks a proper cgroup.

**related**

#1972

### CODE REVIEW

**Review by aiwantaozi — APPROVED** [review]
LGTM

File: pkg/server/runner.go [hunk 1]
```diff
@@ -731,6 +734,49 @@ func (r *Server) configure(_ context.Context) error {
...
+
+			addrs, err := _if.Addrs()
+			if err != nil {
+				return fmt.Errorf("error querying addresses from interface %s(%d): %w",
+					_if.Name, _if.Index, err)
+			}
+
+			for _, addr := range addrs {
+				if ipnet, ok := addr.(*net.IPNet); ok && !ipnet.IP.IsLoopback() {
+					// NB(thxCode): There is no NAT for IPv6,
```
Comment (alexcodelf) [inline pkg/server/runner.go:760]: Will it run in a pure IPv6 environment?

Comment (thxCode) [reply pkg/server/runner.go:760]: there is no "private" ipv6 address.

**Review by alexcodelf — APPROVED** [review]
LGTM

**Pull request merged**